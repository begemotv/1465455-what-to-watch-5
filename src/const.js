export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  FILMS: `/films/`,
  FILMS_ID: `/films/:id`,
  FILMS_ID_REVIEW: `/films/:id/review`,
  LOGIN: `/login`,
  MYLIST: `/mylist`,
  PLAYER: `/player/`,
  PLAYER_ID: `/player/:id`,
  REVIEW: `/review`,
  ROOT: `/`,
  SERVER_ERROR: `/error`,
};

export const ApiRoute = {
  COMMENTS: `/comments/`,
  FAVORITE: `/favorite/`,
  FILMS: `/films/`,
  LOGIN: `/login`,
  PROMO: `/films/promo`,
};

export const ReviewLength = {
  MIN: 50,
  MAX: 400,
};

export const HttpCode = {
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  NOT_FOUND: 404,
  SERVER_ERROR: 500,
};

export const FILMS_INCREMENT_COUNT = 8;

export const DEFAULT_ID = 0;

export const DEFAULT_TAB = `Overview`;

export const ErrorMessage = {
  FETCH_FILM_REVIEWS_ERROR: `Failed fetching film reviews from server`,
  FETCH_FILM_PROMO_ERROR: `Failed fetching promo film from server`,
  LOGIN_ERROR: `Error on login`,
  POST_REVIEW_ERROR: `Failed posting review to server`,
  SERVER_ERROR: `Something went wrong on the server`,
};
