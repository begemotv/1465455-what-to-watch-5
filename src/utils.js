export const nameRating = (rating) => {
  let ratingName = ``;
  switch (true) {
    case rating >= 0 && rating < 3:
      ratingName = `Bad`;
      break;
    case rating >= 3 && rating < 5:
      ratingName = `Normal`;
      break;
    case rating >= 5 && rating < 8:
      ratingName = `Good`;
      break;
    case rating >= 8 && rating < 10:
      ratingName = `Very good`;
      break;
    case rating === 10:
      ratingName = `Awesome`;
      break;
    default:
      ratingName = `Not rated`;
  }
  return ratingName;
};

