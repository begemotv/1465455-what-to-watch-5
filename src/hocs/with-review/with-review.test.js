import React from "react";
import renderer from "react-test-renderer";

import withReview from "./with-review";
import {noop} from "../../utils";
import {DEFAULT_ID} from "../../const";

const MockComponent = () => {
  return (
    <React.Fragment/>
  );
};

const MockComponentHOC = withReview(MockComponent);

test(`withReview renders correctly`, () => {
  const tree = renderer
    .create(
        <MockComponentHOC
          id={DEFAULT_ID}
          onReviewAdd={noop}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
