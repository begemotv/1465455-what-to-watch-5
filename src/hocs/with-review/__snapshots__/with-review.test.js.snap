// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withReview renders correctly 1`] = `null`;
